modules/core/src/types/layer-props.ts

```diff
@@ -75,6 +75,19 @@ export type Texture =
   | HTMLVideoElement
   | ImageBitmap;
 
+/**
+ * Default type of `data` property
+ *
+ * If it's a `string` it's interpreted as URL which passed to `fetch`.
+ * If it's a `LayerData` it is expected it's already resolved.
+ * If it's promise or async iterable resolves it asynchronously.
+ */
+export type LayerDataProp<DataType = any> =
```

Comment:
- Should we use `unknown` instead of `any`? Maybe that would be a breaking change best kept for v9?

```suggestion
export type LayerDataProp<DataType = unknown> =
```